astropy/units/core.py

```diff
@@ -473,7 +473,7 @@ def set_enabled_equivalencies(equivalencies):
         >>> from astropy import units as u
         >>> with u.set_enabled_equivalencies(u.dimensionless_angles()):
         ...     phase = 0.5 * u.cycle
-        ...     np.exp(1j*phase)  # doctest: +FLOAT_CMP
+        ...     np.exp(1j*phase)  # doctest: +FLOAT_CMP +IGNORE_WARNINGS
```

Comment:
Hmm, I'd be interested to know what the actual warning was. I cannot find this in #11821. Is there perhaps a log I can look at? (I've yet to install a beta version of python; if that's easy, I'll do that!)

Reply:
Same problem as all others failures, dividing a complex by a scalar now raises a warning because of the division by `0j`:
```
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
465         list of equivalent pairs, e.g., as returned by
466         `~astropy.units.equivalencies.dimensionless_angles`.
467 
468     Examples
469     --------
470     Exponentiation normally requires dimensionless quantities.  To avoid
471     problems with complex phases::
472 
473         >>> from astropy import units as u
474         >>> with u.set_enabled_equivalencies(u.dimensionless_angles()):
UNEXPECTED EXCEPTION: RuntimeWarning('divide by zero encountered in cdouble_scalars')
Traceback (most recent call last):
  File "/home/sconseil/.pyenv/versions/3.10.0b4/lib/python3.10/doctest.py", line 1348, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest astropy.units.core.set_enabled_equivalencies[1]>", line 3, in <module>
  File "/home/sconseil/dev/astropy/astropy/astropy/units/quantity.py", line 504, in __array_ufunc__
    arrays.append(converter(input_) if converter else input_)
  File "/home/sconseil/dev/astropy/astropy/astropy/units/core.py", line 988, in convert
    return func(_condition_arg(v) / scale1) * scale2
RuntimeWarning: divide by zero encountered in cdouble_scalars
/home/sconseil/dev/astropy/astropy/astropy/units/core.py:474: UnexpectedException
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/sconseil/dev/astropy/astropy/astropy/units/core.py(988)convert()
-> return func(_condition_arg(v) / scale1) * scale2
(Pdb) _condition_arg(v)
0.5j
(Pdb) scale1
0.15915494309189535
(Pdb) _condition_arg(v) / scale1
*** RuntimeWarning: divide by zero encountered in cdouble_scalars
```

Reply:
Is this python 3.10 specific? This is completely unphysical, so I'd want to raise a bug report...